main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -1933,7 +1933,7 @@ object Weeder {
               // Doing a manual flatMap to keep the function tail-recursive
               translateHexCode(code, mkSL(sp1, sp2)) match {
                 case Validation.Success(char) => visit(rest2, char :: acc)
-                case Validation.Failure(errors) => Validation.Failure(errors)
+                case failure => Validation.Failure(failure.errors)
```

Comment:
Should this call recursively on `SuccessWithFailures`? That would, however, require rewriting the function slightly to keep it tail recursive.

Reply:
Fine for now. We can revisit once we start to really use `SuccessWithFailures`